### PR TITLE
Fixed list index out of range error with show arp (#262)

### PR DESCRIPTION
**Pull request by madhupalu**
commit f0520d8016e3a3bca4c4b965bb82a6d5c6fe3166 (HEAD -> show-azure-262, origin/show-azure-262)
Author: madhu Pal <madhupa@aviznetworks.com>
Date:   Sun Feb 3 01:55:26 2019 +0000

    Fixed list index out of range error with show arp (#262)

    Signed-off-by: madhu Pal <madhupa@aviznetworks.com>

**- What I did**
Fixed list index out of range exception in nbrshow while fetching fdb data.
**- How I did it**
Added size check for the data structure (fdb entry) before accessing/updating it.
**- How to verify it**
To verify run as below:
image
image
image
image

**root@sonic:/home/admin# arp -n
Address                  HWtype  HWaddress           Flags Mask            Iface**
10.0.0.1                                   (incomplete)                                        Ethernet0

### CODE REVIEW

**Comment by madhupalu** [comment]
Fixed issue - https://github.com/Azure/SONiC/issues/262

File: show/main.py [hunk 1]
```diff
@@ -366,7 +366,7 @@ def cli():
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
 def arp(ipaddress, iface, verbose):
     """Show IP ARP table"""
-    cmd = "nbrshow -4"
+    cmd = "arp -n"
```
Comment (prsunny) [inline show/main.py]: `nbrshow `is a script that consolidates relevant output with matching FDB entry and displays the port information as well. Whereas `arp `is a linux command. This is not the intention of `show arp` command.

**Comment by madhupalu** [comment]
thanks for the comments, I'll look into it.

**Comment by madhupalu** [comment]
Prince, 
Found the root cause, will raise a PR shortly. 
Thanks,
-Madhu

**Comment by prsunny** [comment]
@madhupalu , Changes looks good, can you provide the show output with and without fix for the issue reported?

**Comment by madhupalu** [comment]
Prince, 
I've updated the show outputs [show arp/nbrshow -4 & nbr show -6]and fixed `show arp` table display as well. 
let me know the comments. 

Thanks,
-Madhu

**Comment by prsunny** [comment]
@madhupalu , I just checked the code and command output, there is a check to see if "`ether`" is present in the "`arp -n`" show. My thinking is, if mac address is not available, then the entries would be skipped, Can you please provide the "`arp -n`" output for the same above set of entries?

**Comment by madhupalu** [comment]
Prince, 
I have updated the `arp -n` output to the PR. In addition to that, as per your comments I used `ether type` check is the right way to address this issue and re tested it. Here is the output. 
**root@sonic:/home/admin# show arp** 
Address    MacAddress    Iface    Vlan
---------  ------------  -------  ------
Total number of entries 0 
**root@sonic:/home/admin# nbrshow -4**
Address    MacAddress    Iface    Vlan
---------  ------------  -------  ------
Total number of entries 0 
**root@sonic:/home/admin# nbrshow -6**
Address                    MacAddress         Iface        Vlan    Status
-------------------------  -----------------  -----------  ------  --------
fe80::2                    52:56:00:00:00:02  Ethernet0    -       STALE
fe80::2                    52:56:00:00:00:02  eth1         -       STALE
fe80::3c0e:96ff:fe17:19da  fe:54:00:5a:a4:f8  Ethernet32   -
fe80::3c0e:96ff:fe17:19da  fe:54:00:5a:a4:f8  eth9         -

Bottomline is - latest sonic utilities seems has `ether check`, no need to merge these changes. thanks.

**Comment by prsunny** [comment]
Closing PR based on test results! Please re-open if required later

**Pull request closed without merging**